src/pudl/transform/eia861.py

```diff
@@ -1258,6 +1273,138 @@ def demand_side_management(tfr_dfs):
+    # * Clean NERC region col
+    # * Drop data_status and demand_side_management (all F) cols
+    ###########################################################################
+
+    transformed_dsm1 = (
+        _clean_nerc(raw_dsm, idx_cols)
+        .drop(['demand_side_management', 'data_status'], axis=1)
+        .query("utility_id_eia not in [88888]")
+    )
+
+    # Separate dsm data into sales vs. other table (the latter of which can be tidied)
+    dsm_sales = transformed_dsm1[idx_cols + sales_cols].copy()
+    dsm_ee_dr = transformed_dsm1.drop(sales_cols, axis=1)
+
+    ###########################################################################
+    # Tidy Data:
+    ###########################################################################
+
+    tidy_dsm, dsm_idx_cols = (
+        pudl.transform.eia861._tidy_class_dfs(
+            dsm_ee_dr,
+            df_name='Demand Side Management',
+            idx_cols=idx_cols,
+            class_list=pc.CUSTOMER_CLASSES,
+            class_type='customer_class',
+            keep_totals=True
+        )
+    )
+
+    ###########################################################################
+    # Transform Data Round 2
+    # * Make booleans (Y=True, N=False)
+    # * Turn 1000s of dollars back into dollars
+    ###########################################################################
+
+    # Split tidy dsm data into transformable chunks
+    tidy_dsm_bool = (
+        tidy_dsm[dsm_idx_cols + bool_cols].copy()
+        .set_index(dsm_idx_cols)
+    )
+    tidy_dsm_cost = (
+        tidy_dsm[dsm_idx_cols + cost_cols].copy()
+        .set_index(dsm_idx_cols)
+    )
+    tidy_dsm_ee_dr = (
+        tidy_dsm.drop(bool_cols + cost_cols, axis=1)
+    )
+
+    # Calculate transformations for each chunk
+    transformed_dsm2_bool = (
+        _make_yn_bool(tidy_dsm_bool)
+        .reset_index()
+        .assign(short_form=lambda x: x.short_form.fillna(False))
+    )
+    transformed_dsm2_cost = _thousand_to_one(tidy_dsm_cost).reset_index()
+
+    # Merge transformed chunks back together
+    transformed_dsm2 = (
+        pd.merge(transformed_dsm2_bool, transformed_dsm2_cost,
+                 on=dsm_idx_cols, how='outer')
+    )
+    transformed_dsm2 = (
+        pd.merge(transformed_dsm2, tidy_dsm_ee_dr,
+                 on=dsm_idx_cols, how='outer')
+    )
+
+    # Split into final tables
+    ee_cols = [col for col in transformed_dsm2 if 'energy_efficiency' in col]
+    dr_cols = [col for col in transformed_dsm2 if 'load_management' in col]
+    program_cols = ['price_responsiveness_customers',
+                    'time_responsiveness_customers']
+    total_cost_cols = ['annual_indirect_program_cost', 'annual_total_cost']
+
+    dsm_program_customers = (
+        transformed_dsm2[dsm_idx_cols + program_cols].copy()
+    )
+    dsm_ee_dr = (
+        transformed_dsm2[
+            dsm_idx_cols
+            + ee_cols
+            + dr_cols
+            + total_cost_cols].copy()
+    )
```

Comment:
I think I separated them for content purposes, but I think for clarity's sake I should combine them.

Reply:
oh, I'm pretty sure it was because I was considering breaking down (wide to tall) the dr ee components of the dr_ee table by their status as either 'energy_efficiency' or 'load_management' columns, specifically. In the raw form of this table, these columns have that as a suffix (the program columns do not) which was why they were in separate tables.

Reply:
great!